### PR TITLE
DOC/RLS: add changelog for 0.10 release

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
In preparation of https://github.com/geopandas/geopandas/issues/2076

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,86 @@
...
 
+Version 0.10.0 (September ??, 2021)
+-----------------------------------
+
+
+Highlights of this release:
+
+- A new `sjoin_nearest()` function to join based on proximity, with the
+  ability to set a maximum search radius (#1865). In addition, the `sindex`
+  attribute gained new methdods for a "nearest" spatial index query (#1865,
```
Comment (martinfleis) [inline CHANGELOG.md:12]:
```suggestion
  attribute gained new methods for a "nearest" spatial index query (#1865,
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,86 @@
...
+- Fix in `GeoDataFrame.apply()` to preserve the active geometry column name
+  (#1955).
+- Fix in `sjoin()` to not ignore the suffixes in case of a right-join
+  (`how="right`) (#2065).
+- Fix `GeoDataFrame.explode()` with a MultiIndex (#1945).
+- Fix the handling of missing values in `to/from_wkb` and `to_from_wkt` (#1891).
+- Fix `to_file()` and `to_json()` when DataFrame has duplicate columns to
+  raise an error (#1900).
+- Fix bug in the colors shown with user-defined classification scheme (#2019).
+- Fix handling of the `path_effects` keyword in `plot()` (#2127).
```
Comment (martinfleis) [inline CHANGELOG.md:69]:
```suggestion
- Fix handling of the `path_effects` keyword in `plot()` (#2127).
- Fix `GeoDataFrame.explode()` to preserve `attrs` (#1935)
```

**Review by martinfleis — COMMENTED** [review]
One typo and one missed bug fix.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,93 @@
...
+  available as methods on the GeoDataFrame (#2141, #1984, #2150).
+
+New features and improvements:
+
+- Add support for pandas' `value_counts()` method for geometry dtype (#2047).
+- The `explode()` method has a new `ignore_index` keyword (consistent with
+  pandas' explode method) to reset the index in the result, and a new
+  `index_parts` keywords to control whether a cumulative count indexing the
+  parts of the exploded multi-geometries should be added (#1871).
+- `points_from_xy()` is now available as a GeoSeries method (#1936).
```
Comment (martinfleis) [inline CHANGELOG.md:25]:
```suggestion
- `points_from_xy()` is now available as a GeoSeries method `from_xy` (#1936).
```

Just a minor change to make it explicit that the method is not `GeoSeries.points_from_xy` but `GeoSeries.from_xy`.

Comment (jorisvandenbossche) [reply CHANGELOG.md:25]: Hmm, I didn't actually realize that in the PR that added this. I understand that the shorter method name might be a bit nicer, but the inconsistent naming between the function and the method is also a bit unfortunate ...

Comment (martinfleis) [reply CHANGELOG.md:25]: I think it was actually my proposal, to keep it consistent with the `GeoSeries.from_*` bunch of methods.

Comment (jorisvandenbossche) [reply CHANGELOG.md:25]: Yes, there is also the consistency with that :)

**Comment by jorisvandenbossche** [comment]
Merge this and push the release button?

**Comment by martinfleis** [comment]
> push the release button

Let me check how docs build before pushing the button.

**Comment by jorisvandenbossche** [comment]
The last one was passing: https://readthedocs.org/projects/geopandas/builds/

**Comment by martinfleis** [comment]
There's a warning in the RTD log about `nearest_all` not being found but I am not sure where does it come from. There's no mention of it in the relevant code.

```py
WARNING: autodoc: failed to import method 'SpatialIndex.nearest_all' from module 'geopandas.sindex'; the following exception was raised:
Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/geopandas/conda/latest/lib/python3.9/site-packages/sphinx/util/inspect.py", line 423, in safe_getattr
    return getattr(obj, name, *defargs)
AttributeError: type object 'SpatialIndex' has no attribute 'nearest_all'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/docs/checkouts/readthedocs.org/user_builds/geopandas/conda/latest/lib/python3.9/site-packages/sphinx/ext/autodoc/importer.py", line 110, in import_object
    obj = attrgetter(obj, mangled_name)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geopandas/conda/latest/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 332, in get_attr
    return autodoc_attrgetter(self.env.app, obj, name, *defargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geopandas/conda/latest/lib/python3.9/site-packages/sphinx/ext/autodoc/__init__.py", line 2771, in autodoc_attrgetter
    return safe_getattr(obj, name, *defargs)
  File "/home/docs/checkouts/readthedocs.org/user_builds/geopandas/conda/latest/lib/python3.9/site-packages/sphinx/util/inspect.py", line 439, in safe_getattr
    raise AttributeError(name) from exc
AttributeError: nearest_all
```

**Comment by martinfleis** [comment]
I think we can safely ignore it.

**Pull request merged**